genie-agent/src/main/java/com/netflix/genie/agent/AgentMetadataImpl.java

```diff
@@ -38,17 +36,18 @@
  * @since 4.0.0
  */
 @Slf4j
-@Component
 @Getter
-@Lazy
-class AgentMetadataImpl implements AgentMetadata {
+public class AgentMetadataImpl implements AgentMetadata {
```

Comment:
Kinda sucks that we have to make all classes public because configs are in a different package... Not much to do about it i guess. Probably not worth scattering configs all over.

Reply:
So you're ok with me making the classes public and centralizing the configuration classes? I didn't do this cause I didn't know how you'd feel about it but I wanted to. Right now the configuration classes are spread all over and they could be centralized and consolidated if we made classes public

Reply:
Just to make sure we're on the same page..

Option 1 - Leave implementations (i.e. most stuff that was tagged @Component earlier) package-protected, create a configuration for those beans in the same package.

Option 2 - Make the implementations public. Centralize all configurations in the `config` package

(there's more options but probably not even worth bringing up..)

I can see upside and downside to both options. 
Personally I prefer \#2, but since the convention in this repo is already \#1 i would say let's keep going with that. No strong opinion either way.

Reply:
The second option is also what I'm favoring and is how we do it in the `web` project so it would bring consistency in that respect.